### ISSUE TITLE
PostgreSQL gset causes next line to fail to parse

### ISSUE THREAD

**Issue by jambonrose** [sqlfluff/sqlfluff#7172]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Hi, thanks for making and maintaining SQLFluff.

I'm running into an issue with PostgreSQL's `gset`. I have code that runs without issue (I'm running tests via pgTAP), but SQLFluff says that it is unable to parse it.

### Expected Behaviour

I would have expected the linter to parse the code.

### Observed Behaviour

The example below fails with the following:

```
== [stdin] FAIL                                                                                                                 
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'UPDATE
                       | center_phone_number\nSET number = ...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```shell 
cat << 'EOF' | sqlfluff lint --dialect postgres -
SELECT id AS test_id
FROM center
WHERE slug = 't6pggzLNq9zL'
\gset

UPDATE center_phone_number
SET number = '555-UPDATED'
WHERE center_id = :'test_id';
EOF
```

### Dialect

postgres

### Version

```console
$ sqlfluff --version
sqlfluff, version 3.4.2
$ python --version
Python 3.13.7
$ sw_vers
ProductName:		macOS
ProductVersion:		15.7.1
BuildVersion:		24G231
```

### Configuration

N/A - I run the command above without a config file.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
FIxed with #6555
